src/optimistic-oracle/optimistic-oracle.js

```diff
@@ -163,26 +148,27 @@ function provideHandleTransaction(erc20Contract) {
       }
 
       if (log.name === 'ProposePrice') {
-        const { currency } = log.args;
-        const { requester } = log.args;
-        const { proposer } = log.args;
+        const {
+          currency, requester, proposer, proposedPrice: proposedPriceRaw,
+        } = log.args;
+        let erc20Contract;
 
         // set up ERC-20 contract to get decimals value
         // in production, erc20Contract will always be undefined
         // in testing, erc20Contract will be assigned to a mock contract
-        /* eslint-disable no-param-reassign */
-        if (erc20Contract === undefined) {
+        if (erc20MockContract === undefined) {
           erc20Contract = new ethers.Contract(currency, erc20Abi, provider);
+        } else {
+          erc20Contract = erc20MockContract;
         }
-        /* eslint-enable no-param-reassign */
 
         const decimals = await erc20Contract.decimals();
 
         // convert proposedPrice to a human-readable decimal value
-        let proposedPrice = ethers.utils.formatUnits(log.args.proposedPrice, decimals);
+        const proposedPrice = parseFloat(ethers.utils.formatUnits(proposedPriceRaw, decimals));
 
         // convert the proposedPrice to a BigNumber type for difference calculations later
-        proposedPrice = new BigNumber(proposedPrice.toString());
+        const proposedPriceBN = new BigNumber(proposedPrice.toString());
```

Comment:
this is backwards. don't use `parseFloat`, if you find yourself using `parseInt` and `parseFloat` in javascript for ethereum you are likely doing it wrong
```
> (new bn("1.11111111111111111111111111111111111")).toString()
'1.11111111111111111111111111111111111'
> parseFloat("1.11111111111111111111111111111111111").toString()
'1.1111111111111112'
```
you can lose percision going from string -> number but not from string -> bignubmer

Reply:
I removed all use of parseFloat